clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1215,6 +1241,69 @@ class ArchiveFileHandler final : public FileHandler {
   Error WriteBundle(raw_fd_ostream &OS, MemoryBuffer &Input) override {
     llvm_unreachable("unsupported for the ArchiveFileHandler");
   }
+
+private:
+  // NOTE: mostly a copy-paste of ReadHeader method.
+  Expected<std::vector<std::string>>
+  ReadTargetsFromChild(const Archive::Child &C) {
+    auto BinOrErr = C.getAsBinary();
+    if (!BinOrErr)
+      return BinOrErr.takeError();
+
+    auto &Bin = BinOrErr.get();
+    auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
+    auto Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
+    ObjectFileHandler OFH(std::move(Obj), BundlerConfig);
+    if (Error Err = OFH.ReadHeader(*Buf))
+      return Err;
+    Expected<std::optional<StringRef>> NameOrErr = OFH.ReadBundleStart(*Buf);
+    if (!NameOrErr)
+      return NameOrErr.takeError();
+
+    std::vector<std::string> Targets;
+    while (*NameOrErr) {
+      if (*NameOrErr)
+        Targets.emplace_back((**NameOrErr).str());
+      NameOrErr = OFH.ReadBundleStart(*Buf);
+      if (!NameOrErr)
+        return NameOrErr.takeError();
+    }
+
+    return Targets;
+  }
+
+  bool CheckTargetIsExcluded(StringRef Triple) {
+    // NOTE: "-sycldevice" Triple component has been deprecated.
+    // However, it still can be met in libraries that have been compiled before
+    // deprecation. For example, here Triple might be the following:
+    //  sycl-fpga_aoco-intel-unknown-sycldevice
+    //
+    // The workaround is to strip this Triple component if it is present.
+    Triple.consume_back("-sycldevice");
+    const auto &ExcludedTargetNames = BundlerConfig.ExcludedTargetNames;
+    auto it = std::find(ExcludedTargetNames.begin(), ExcludedTargetNames.end(),
+                        Triple);
+    return it != ExcludedTargetNames.end();
```

Comment:
```suggestion
    return std::count_if(ExcludedTargetNames.begin(), ExcludedTargetNames.end(),
                        Triple) > 0;
```

Feel free to ignore, though, but `it` should be replaced with `It` in that case to conform with coding standards